src/app/components/utility/modal/modal.component.ts

```diff
@@ -230,33 +235,80 @@ export class ModalComponent implements OnInit {
+    return (res !== null);
   }
 
+   validateModalInput(e) {
+
+     this.inputErrorMessage = '';
+     if (e.target.name === 'challegenDeleteInput') {
+       this.isDisabled = e.target.value !== this.challenge.title;
+     } else if (e.target.name === 'editChallengeTitle') {
+       this.isDisabled = e.target.value === this.challenge.title;
+     } else if (e.target.name === 'update_first_name') {
+       this.isDisabled = e.target.value === this.user.first_name;
+     } else if (e.target.name === 'update_last_name') {
+       this.isDisabled = e.target.value === this.user.last_name;
+     } else if (e.target.name === 'update_affiliation') {
+       this.isDisabled = e.target.value === this.user.affiliation;
+     } else if (e.target.name === 'update_google_scholar_url') {
+       if (this.validProtocol(e.target.value) && this.validURL(e.target.value)) {
+         this.isDisabled = e.target.value === this.user.google_scholar_url;
+
+         this.invalidFields = this.invalidFields.filter(element => element !== 'Google Scholar');
+         this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? (this.invalidFields.length > 2 ? `${this.invalidFields.join(', ')} are` : `${this.invalidFields.join(' and ')} are`) : `${this.invalidFields.join('')} is`} not valid` : null;
+       } else {
+         this.isDisabled = false; // In case they want to empty the field.
+
+         if (!(this.invalidFields.find(element => element === 'Google Scholar'))) { this.invalidFields.push('Google Scholar'); }
+         this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? (this.invalidFields.length > 2 ? `${this.invalidFields.join(', ')} are` : `${this.invalidFields.join(' and ')} are`) : `${this.invalidFields.join('')} is`} not valid` : null;
+       }
+     } else if (e.target.name === 'update_github_url') {
+      if (this.validProtocol(e.target.value) && this.validURL(e.target.value)) {
+        this.isDisabled = e.target.value === this.user.github_url;
+
+        this.invalidFields = this.invalidFields.filter(element => element !== 'GitHub');
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? (this.invalidFields.length > 2 ? `${this.invalidFields.join(', ')} are` : `${this.invalidFields.join(' and ')} are`) : `${this.invalidFields.join('')} is`} not valid` : null;
+       } else {
+        this.isDisabled = false; // In case they want to empty the field.
+
+        if (!(this.invalidFields.find(element => element === 'GitHub'))) { this.invalidFields.push('GitHub'); }
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? (this.invalidFields.length > 2 ? `${this.invalidFields.join(', ')} are` : `${this.invalidFields.join(' and ')} are`) : `${this.invalidFields.join('')} is`} not valid` : null;
```

Comment:
This doesn't look good. Can we simplify this expression? or move it in a separate place and then call it from here.